jdaviz/configs/mosviz/helper.py

```diff
@@ -528,19 +583,196 @@ def load_images(self, data_obj, data_labels=None, share_image=0):
+            If False: will only show column names currently not shown in the table.
+        """
+        if visible is None:
+            return [c.label for c in self.app.data_collection['MOS Table'].components]
+        elif visible is True:
+            return [h['value'] for h in self.app.get_viewer('table-viewer').widget_table.headers]
+        elif visible is False:
+            return [cn for cn in self.get_column_names() if cn not in self.get_column_names(True)]
+        else:
+            raise ValueError("visible must be one of None, True, or False.")
+
+    def set_visible_columns(self, column_names=None):
+        """
+        Set the columns to be visible in the table.
+
+        Parameters
+        ----------
+        column_names: list or None
+            list of columns to be visible in the table.  If None, will default to original
+            visible columns.
+        """
+        if column_names is None:
+            column_names = self._default_visible_columns
+        if not isinstance(column_names, list):
+            raise TypeError("column_names must be of type list")
+        avail_names = self.get_column_names()
+        if not np.all([c in avail_names for c in column_names]):
+            raise ValueError("not all entries of column_names are valid")
+        is_sortable = ['Redshift']
+        headers = [{'text': cn, 'value': cn, 'sortable': cn in is_sortable} for cn in column_names]
+        wt = self.app.get_viewer('table-viewer').widget_table
+        wt.set_state({'headers': headers})
+        wt.send_state()
+
+    def hide_column(self, column_name):
+        """
+        Hide a single column in the table.
+
+        Parameters
+        ----------
+        column_name: str
+            Name of the column to hide
+        """
+        if not isinstance(column_name, str):
+            raise TypeError("column_name must be of type str")
+
+        column_names = self.get_column_names()
+        if column_name not in column_names:
+            raise ValueError(f"{column_name} not in available columns ({column_names})")
+        new_column_names = [cn for cn in self.get_column_names(True)
+                            if cn not in column_name]
+        return self.set_visible_columns(new_column_names)
+
+    def show_column(self, column_name):
+        """
+        Show a hidden column in the table.
+
+        Parameters
+        ----------
+        column_name: str
+            Name of the column to show
+        """
+        if not isinstance(column_name, str):
+            raise TypeError("column_name must be of type str")
+
+        column_names = self.get_column_names(True)
+        if column_name not in column_names:
+            if column_name in self.get_column_names():
+                return self.set_visible_columns(column_names+[column_name])
+            else:
+                raise ValueError(f"{column_name} not in available columns ({column_names})")
```

Comment:
This error only suggests column names that are already visible, which is not super helpful because when you are calling `show_column`, you want to show something that is not visible yet.

Reply:
good catch!  See 3e17c66